services/navigation/builtin/builtin.go

```diff
@@ -93,16 +95,30 @@ func (conf *Config) Validate(path string) ([]string, error) {
 	}
 	deps = append(deps, conf.BaseName)
 
-	if conf.MovementSensorName == "" {
-		return nil, utils.NewConfigValidationFieldRequiredError(path, "movement_sensor")
-	}
-	deps = append(deps, conf.MovementSensorName)
-
 	if conf.MotionServiceName == "" {
 		conf.MotionServiceName = resource.DefaultServiceName
 	}
 	deps = append(deps, resource.NewName(motion.API, conf.MotionServiceName).String())
 
+	if conf.MapTypeName == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "map_type")
+	}
+
+	if !slices.Contains([]string{"None", "GPS"}, conf.MapTypeName) {
+		return nil, errors.New("invalid map_type, map_type must be one of the following ['None', 'GPS']")
+	}
+
+	if conf.MapTypeName == "GPS" {
+		if conf.MovementSensorName == "" {
+			return nil, utils.NewConfigValidationFieldRequiredError(path, "movement_sensor")
+		}
+		deps = append(deps, conf.MovementSensorName)
+	}
+
+	if conf.MapTypeName == "None" && len(conf.VisionServices) == 0 {
```

Comment:
[note] How/when do we want to check if a vision service is configured? Either require it always or only check when setting mode to explore.